protobuf/flow/entities/block.proto

```diff
@@ -12,6 +12,13 @@ import "flow/entities/block_seal.proto";
 import "flow/entities/execution_result.proto";
 import "flow/entities/block_header.proto";
 
+enum BlockStatus {
+  BLOCK_UNKNOWN = 0;
+  BLOCK_FINALIZED = 1;
+  BLOCK_EXECUTED = 2;
```

Comment:
@lolpuddle when would a block be executed but not finalized or sealed?

Reply:
I guess realistically never because we'd probably return a storage error if we couldn't find the block, and if we were able to find the block locally then it would either be finalized, or sealed. I'll remove the finalized state